### ISSUE TITLE
Expose HAProxy stats to prometheus

### ISSUE THREAD

**Issue by TamalSaha** [appscode/voyager#13]
https://github.com/appscode/haproxy_exporter

@sadlil , I think we can take the same approach we are taking with k8sdb regarding exporters. A single exporter running in the same namespace as the operator can take care of exposing these metrics for all Haproxys.

**Comment by aerokite** [comment]
sidecar container for exporter will be easier

**Comment by TamalSaha** [comment]
```
ingress.appscode.com/stats: true   => ingress-stats
ingress.appscode.com/stats.secretName: name  => auth for stats endpoint
ingress.appscode.com/stats.port: 1936
ingress.appscode.com/stats.serviceName: <ingress>-stats

ingress.appscode.com/stats.prometheusLabels: {"k": "v"}

// Exporter Deployment
// Exporter Service
// ServiceMonitor with label
```

**Comment by TamalSaha** [comment]
Closed in favor of https://github.com/appscode/voyager/issues/106

**Comment by TamalSaha** [comment]
To export HAProxy stats to Prometheus user can apply the following annotations to an `ingress` object.
```
monitoring.appscode.com/agent: Prometheus

monitoring.appscode.com/prometheus.Namespace: <Kube NS where service monitors will be created>
monitoring.appscode.com/prometheus.Labels: <map[string]string used to select Prometheus instance>
monitoring.appscode.com/prometheus.Interval: <scrape interval>
```

Voyager operator will automatically create `ServiceMonitor` TPR objects based on these annotations. That means if you also use Prometheus operator, your HAProxy stats will start showing up in Prometheus and  Grafana Dashboard without further manual action.

**Comment by TamalSaha** [comment]
Fixed in #153
